src/main/kotlin/blackjack/controller/Controller.kt

```diff
@@ -0,0 +1,98 @@
+package blackjack.controller
+
+import blackjack.model.CardDeck
+import blackjack.model.Dealer
+import blackjack.model.ErrorMessage
+import blackjack.model.Player
+import blackjack.model.Players
+import blackjack.model.ResultEvaluator
+import blackjack.view.InputView
+import blackjack.view.OutputView
+import blackjack.view.OutputView.displayInitialCardsMessage
+
+class Controller {
+    private val deck = CardDeck()
+    private var _players: Players? = null
+    private val dealer = Dealer()
+
+    private val players: Players
+            get() = _players!!
+
+    fun runGame() {
+        _players = initializePlayers()
+        OutputView.displayPlayerNames(players)
+        drawInitialCards()
+        OutputView.displayFirstCardMessage(dealer)
+        players.forEach { OutputView.displayAllCardsMessage(it) }
+        players.dealCards()
+        dealer.dealCards()
+        displayCardsAndTotal()
+        printResults()
+    }
+
+    fun printResults() {
+        val evaluator = ResultEvaluator(players, dealer)
+        val results = evaluator.calculateResults()
+        OutputView.displayResults(results)
+    }
+
+    fun displayCardsAndTotal() {
+        OutputView.displayParticipantStatus(dealer)
+        players.forEach { OutputView.displayParticipantStatus(it) }
+    }
+
+    fun Dealer.dealCards() {
+        while (this.shouldDraw()) {
+            OutputView.displayDealerDrawMessage()
+            this.drawCard(deck)
+        }
+    }
+
+    fun Players.dealCards() {
+        players.forEach { dealCards(it) }
+    }
+
+    fun dealCards(player: Player) {
+        while (player.isNotBusted() && wantsToDraw(player)) {
+            player.drawCard(deck)
+            OutputView.displayAllCardsMessage(player)
+        }
+    }
+
+    fun wantsToDraw(player: Player): Boolean {
+        repeat(MAX_TRIES){
+            try {
+                return InputView.promptForDraw(player)
+            } catch (e: IllegalArgumentException) {
+                println(e.message)
```

Comment:
Although this is a simple logic, I think view related classes should take care of printing out errors as well. Give them the exception, messages, causes, or anything, and let them take care of printing what they want. How does this sound?

Reply:
If we extend this idea, repeating `MAX_TRIES` until we get the desired input can be done in view as well. Controller and domain related logics don't need to know how many tries were taken before we actually play blackjack. They just want to play the game with nice player names and their 'yes's and 'no's. :)